src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -139,6 +142,38 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
         raise CLIError('Either the kafka client group id and kafka client group name should be specified, '
                        'or none of them should be')
 
+    # Validate and initialize autoscale setting
+    autoscale_configuration = None
+    if autoscale_type == 'Load':
+        if not all([autoscale_min_count, autoscale_max_count]):
+            raise CLIError(
+                'When the autoscale type is Load, both of the autoscal min count and max count should be specified.')
+
+        autoscale_configuration = Autoscale(
+            capacity=AutoscaleCapacity(
+                min_instance_count=autoscale_min_count,
+                max_instance_count=autoscale_max_count
+            )
+        )
+    elif autoscale_type == 'Schedule':
+        if not all([timezone, days, time, autoscale_count]):
+            raise CLIError(
+                'When the autoscale is Schedule, all of the timezone, days, time, autoscale_count should be specified.')
```

Comment:
>autoscale [](start = 26, length = 9)

1 ...autoscale type is ...
2  In CLIError message, using parameter exposed to user is better. We can ask Dongwei for his opinion.

Reply:
updated.

---
In reply to: [466795839](https://github.com/Azure/azure-cli/pull/14692#discussion_r466795839) [](ancestors = 466795839)